javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserClassDeclaration.java

```diff
@@ -465,17 +465,23 @@ private ResolvedReferenceType toReferenceType(ClassOrInterfaceType classOrInterf
             // look for the qualified name (for example class of type Rectangle2D.Double)
             className = classOrInterfaceType.getScope().get().toString() + "." + className;
         }
-        SymbolReference<ResolvedTypeDeclaration> ref = solveType(className);
+
```

Comment:
Please comment on why, at this stage, type resolution should be delegated to the parent context.

Reply:
toReferenceType is used to resolve reference to "extends" and "implements" types. These types should not be resolved against e.g types with the same name defined within the class.
As per the test case I included -- 

```
class A extends B { 
static class B ... {} 
}

```
The B in "extends B" should not resolve to the B in "static class B"... Hence we should resolve based on the context containing the class.. Does this make sense?

Reply:
I meant adding comments to the code.

Reply:
Added, thanks